mne/viz/epochs.py

```diff
@@ -65,74 +78,298 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
+        Arguments passed to a call to `mne.viz.plot_compare_evoked` to style
+        the evoked plot below the image. Defaults to an empty dictionary,
+        meaning `plot_compare_evokeds` will be called with default parameters.
 
     Returns
     -------
     figs : lists of matplotlib figures
         One figure per channel displayed.
     """
-    from scipy import ndimage
     units = _handle_default('units', units)
     scalings = _handle_default('scalings', scalings)
 
-    import matplotlib.pyplot as plt
+    # setting defaults
+    if group_by is not None and combine is None:
+        combine = 'gfp'
+
+    if all(param is None for param in (group_by, picks, combine)):
+        group_by = "type"
+        combine = "gfp"
+
     if picks is None:
         picks = pick_types(epochs.info, meg=True, eeg=True, ref_meg=False,
-                           exclude='bads')[:5]
+                           exclude='bads')
+        if group_by is None:
+            picks = picks[:5]  # take 5 picks to prevent spawning many figs
+    else:
+        picks = np.atleast_1d(picks)
+
+    manual_ylims = "ylim" in ts_args
+
+    # input checks
+    if (combine is None and (fig is not None or axes is not None) and
+            len(picks) > 1):
+        raise ValueError('Only single pick can be drawn to a figure/axis; '
+                         'provide only one pick, or use `combine`.')
 
     if set(units.keys()) != set(scalings.keys()):
         raise ValueError('Scalings and units must have the same keys.')
 
-    picks = np.atleast_1d(picks)
-    if (fig is not None or axes is not None) and len(picks) > 1:
-        raise ValueError('Only single pick can be drawn to a figure.')
-    if axes is not None:
-        if fig is not None:
-            raise ValueError('Both figure and axes were passed, please'
-                             'decide between the two.')
-        from .utils import _validate_if_list_of_axes
-        oblig_len = 3 if colorbar else 2
-        _validate_if_list_of_axes(axes, obligatory_len=oblig_len)
-        ax1, ax2 = axes[:2]
-        # if axes were passed - we ignore fig param and get figure from axes
-        fig = ax1.get_figure()
-        if colorbar:
-            ax3 = axes[-1]
-    evoked = epochs.average(picks)
-    data = epochs.get_data()[:, picks, :]
-    n_epochs = len(data)
-    data = np.swapaxes(data, 0, 1)
-    if sigma > 0.:
-        for k in range(len(picks)):
-            data[k, :] = ndimage.gaussian_filter1d(
-                data[k, :], sigma=sigma, axis=0)
+    ch_types = [channel_type(epochs.info, idx) for idx in picks]
+    if len(set(ch_types)) > 1 and group_by is None and combine is not None:
+        warn("Combining over multiple channel types. "
+             "Please use `group_by`.")
+    for ch_type in ch_types:
+        if ch_type not in scalings:
+            # We know it's not in either scalings or units since keys match
+            raise KeyError('%s type not in scalings and units' % ch_type)
```

Comment:
I didn't actually write this - but as the comment says, scalings.keys() == units.keys() right? (see line 177)